### PR TITLE
Add support for LANL IC Machines (Mustang and Wolf)

### PR DESCRIPTION
**Pull request by douglasjacobsen**
This merge adds machine files for LANL IC Machines mustang and wolf.
Additionally, it defines the moab batch system, as it's slightly
different than the default pbs batch system and the modifications are
necessary to run on the local machines.

### CODE REVIEW

**Comment by douglasjacobsen** [comment]
@jnjohnsonlbl, @jgfouca, and @rljacob: This PR is dependent on: https://github.com/ESMCI/cime/pull/14

Feel free to merge this whenever that is present in our master branch.

File: machines-acme/config_machines.xml [hunk 1]
```diff
@@ -763,7 +763,7 @@
          <DOUT_S_ROOT>$CESMSCRATCHROOT/archive/$CASE</DOUT_S_ROOT>               <!-- complete path to a short term archiving directory -->
          <DOUT_L_MSROOT>USERDEFINED_optional_run</DOUT_L_MSROOT>           <!-- complete path to a long term archiving directory -->
          <CCSM_BASELINE>/projects/ccsm/ccsm_baselines</CCSM_BASELINE>
-         <CCSM_CPRNC>$CIMEROOT/tools/cprnc/build/cprnc</CCSM_CPRNC>                <!-- path to the cprnc tool used to compare netcdf history files in testing -->
+         <CCSM_CPRNC>$CCSMROOT/tools/cprnc/build/cprnc</CCSM_CPRNC>                <!-- path to the cprnc tool used to compare netcdf history files in testing -->
```
Comment (rljacob) [inline machines-acme/config_machines.xml]: @douglasjacobsen can you remove this change?  @jgfouca wants it as CIMEROOT.

File: machines-acme/Makefile [hunk 2]
```diff
@@ -690,9 +683,6 @@ $(EXEC_SE): $(OBJS) $(ULIBDEP) $(CSMSHARELIB) $(MCTLIBS) $(PIOLIB) $(GPTLLIB)
 $(COMPLIB): $(OBJS) 
 	$(AR) -r $(COMPLIB) $(OBJS)
 
-$(LIBROOT)/libcvmix.a:
-	$(MAKE) -f $(CIMEROOT)/../components/pop/source/cvmix/Makefile SRC_DIR=$(CIMEROOT)/../components/pop/source/cvmix FC=$(SFC) FCFLAGS="$(FFLAGS) $(FREEFLAGS)" LIB_DIR=$(LIBROOT)
-
 .c.o:
 	$(CC) -c $(INCLDIR) $(INCS) $(CFLAGS)  $<
 .F.o:
```
Comment (rljacob) [inline machines-acme/Makefile:698]: @douglasjacobsen can you remove these changes to the Makefile?  They'll come in trough another branch.

**Comment by douglasjacobsen** [comment]
@rljacob I've made the changes you requested. Let me know if there is anything else you see that needs to change.

**Pull request merged**